app/controllers/api/account.php

```diff
@@ -1677,6 +1677,96 @@
+    ->inject('dbForProject')
+    ->action(function (string $targetId, string $providerType, string $identifier, string $providerId, string $name, Event $queueForEvents, Document $user, Request $request, Response $response, Database $dbForProject) {
+        $targetId = $targetId == 'unique()' ? ID::unique() : $targetId;
+
+        $provider = new Document();
+
+        if ($providerType === 'push') {
+            $provider = Authorization::skip(fn () => $dbForProject->getDocument('providers', $providerId));
+
+            if ($provider->isEmpty()) {
+                throw new Exception(Exception::PROVIDER_NOT_FOUND);
+            }
+        }
+
+        if ($user->isEmpty()) {
+            throw new Exception(Exception::USER_NOT_FOUND);
+        }
+
+        $target = Authorization::skip(fn () => $dbForProject->getDocument('targets', $targetId));
+
+        if (!$target->isEmpty()) {
+            throw new Exception(Exception::USER_TARGET_ALREADY_EXISTS);
+        }
+
+        $detector = new Detector($request->getUserAgent());
+        $detector->skipBotDetection(); // OPTIONAL: If called, bot detection will completely be skipped (bots will be detected as regular devices then)
+
+        $os = $detector->getOS();
+        $client = $detector->getClient();
+        $device = $detector->getDevice();
+
+        try {
+            $target = $dbForProject->createDocument('targets', new Document([
+                '$id' => $targetId,
+                '$permissions' => [
+                    Permission::read(Role::user($user->getId())),
+                    Permission::update(Role::user($user->getId())),
+                ],
+                'providerId' => $providerId ?? null,
+                'providerInternalId' => $provider->getInternalId() ?? null,
+                'providerType' =>  $providerType,
+                'userId' => $user->getId(),
+                'userInternalId' => $user->getInternalId(),
+                'identifier' => $identifier,
+                'name' => $name ?? [
+                    'osCode' => $os['osCode'],
+                    'osName' => $os['osName'],
+                    'osVersion' => $os['osVersion'],
+                    'clientType' => $client['clientType'],
+                    'clientCode' => $client['clientCode'],
+                    'clientName' => $client['clientName'],
+                    'clientVersion' => $client['clientVersion'],
+                    'clientEngine' => $client['clientEngine'],
+                    'clientEngineVersion' => $client['clientEngineVersion'],
+                    'deviceName' => $device['deviceName'],
+                    'deviceBrand' => $device['deviceBrand'],
+                    'deviceModel' => $device['deviceModel']
+                ]
+            ]));
+        } catch (Duplicate) {
+            throw new Exception(Exception::USER_TARGET_ALREADY_EXISTS);
+        }
+        Authorization::skip(fn () => $dbForProject->deleteCachedDocument('users', $user->getId()));
```

Comment:
We can remove the auth skip for cache deletes

Reply:
removed these